txp/Pos/Txp/Arbitrary.hs

```diff
@@ -51,9 +63,16 @@ instance Arbitrary TxInWitness where
         ScriptWitness <$> arbitrary <*> arbitrary,
         RedeemWitness <$> arbitrary <*> arbitrary,
         UnknownWitnessType <$> choose (3, 255) <*> scale (min 150) arbitrary ]
+    shrink = genericShrink
```

Comment:
Invariants. I see `choose (3, 255)` for `UnknownWitnessType` above in the `arbitrary` method impl. How do you know that it's not violated by `genericShrink` which would try to bring this numeric value down to `0`?